allennlp/modules/token_embedders/embedding.py

```diff
@@ -180,9 +188,33 @@ def extend_vocab(self,  # pylint: disable=arguments-differ
             vocab_namespace = "tokens"
             logging.warning("No vocab_namespace provided to Embedder.extend_vocab. Defaulting to 'tokens'.")
 
+        extended_num_embeddings = extended_vocab.get_vocab_size(vocab_namespace)
+        if extended_num_embeddings <= self.num_embeddings:
+            # It's already been extended. No need to initialize / read pretrained file in first place (no-op)
+            return
+
+        if self._pretrained_file and os.path.exists(self._pretrained_file):
```

Comment:
is this the right check? couldn't `self._pretrained_file` be like a path to S3? in which case it wouldn't `os.path.exist`?

Reply:
I don't think we explicitly worry about URLs not existing anywhere else, so I'd rather not complicate the code here either. Just assume it's OK if it's a URL, and make sure it exists if it's a file.

Reply:
Sounds good!

Reply:
done. 👍